File: app/src/main/java/com/punchlines/punchline/dagger/PunchlineActivity.java
package com.punchlines.punchline.dagger;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public abstract class PunchlineActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        inject(DaggerPunchlineComponent.create());
    }

    protected abstract void inject(PunchlineComponent component);

}

File: app/src/main/java/com/punchlines/punchline/common/dagger/PunchlineActivity.kt
package com.punchlines.punchline.common.dagger

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

abstract class PunchlineActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        inject(DaggerPunchlineComponent.create())
    }

    protected abstract fun inject(component: PunchlineComponent)

}